### DISCUSSION TITLE
After update issues with Shelly RGBW2 and DUO

### DISCUSSION THREAD

**Discussion by InventoryAndi** [mongoose-os-apps/shelly-homekit#1588] [Q&A] [answered]
Hi everyone. I have had this problem several times now, after updates, recently when updating to firmware 2.13.3
Some Shelly (not all) RGBW 2 and DUO flash when switched on. So the light is e.g. at 100% (percentage does not matter, occurs regardless of the dimming) and then flickers strongly. I mainly use my RGBWs with the CCT setting (but I also have one with the White Mode setting that has the same problem). 
As soon as I open the IP address of the Shelly, it catches and lights up normally. After a while (approx. 5-10 minutes) it starts all over again and starts flickering. Same game. If I then open the IP address again to see the Shelly, it catches again, etc.
Originally I thought that the transformer was fucked and that's where the flickering came from. But replacing the transformer didn't help.
I have often had this problem after firmware updates, but strangely not with all Shellys.
My solution (which worked in 90% of cases) was to revert to stock and then reflash. Most Shellys then worked normally again (sometimes I had to do this 2-3 times for one or the other Shelly until it worked). I had always convinced myself that something hadn't worked properly when updating, which is probably nonsense.

It's just super annoying because you have to reconnect the Shelly to HomeKit after the revert to stock and replace all automations with the new device (which can be quite time-consuming depending on the scope). 

So as I said usually I was able to fix it with the revert to stock and new flashing but as I said it happens actually after every firmware update and HomeKit automations are getting more and more complex so more and more effort to add the new flashed device. Maybe someone has made the same experience?

**Comment by InventoryAndi** [comment]
[IMG_0915_1k.mov.zip](https://github.com/user-attachments/files/18062324/IMG_0915_1k.mov.zip)
Video of Shelly DUO RGBW (the behavior is the same for RGBW 2)

**Comment by markirb** [comment]
Wow that looks strange.

Mh I have a RGBW2 in everyday use with CCT and have non of this at current firmware.

Power Saving is @2. I would not know why behavior should be different on other devices.  I looks like hardware timers get a little bit shaky.... :(

**Reply by InventoryAndi** [reply, marked as answer]
Ah I found the issue. I changed everywhere power saving to 1. I changed it back to 2 and now they run smoothly. No issues anymore. Probably I missed that for RGBW2 and DUO power savings should not be 1. Just in case someone else should have this issue, that is the solution :)

**Comment by TZ2626** [comment]
Hi, 
Thanks for sharing this!
I am about to try to homekit-ify my RGBW2. I'd be shaking a lot less if I had the original fw. (it's not listed in the original thread) Do you happen to have the original fw?

**Reply by markirb** [reply]
Which thread are you referring to? Stock fws are accessible and revert works.

**Reply by TZ2626** [reply]
[reverting to stock firmware](https://github.com/mongoose-os-apps/shelly-homekit/wiki/Flashing#updating-from-stock-firmware)
